### DISCUSSION TITLE
Space Utility considers absolutely positioned child

### DISCUSSION THREAD

**Discussion by goela268** [tailwindlabs/tailwindcss#11711] [Help]
I don't know if this Is considered a bug and there is a relatively easy workaround, but for me this should not be intended behaviour. 

What do you think?

Since the Tailwind playground returns an Error 500 when you click on share, here is my code to reproduce
```
<div class="p-8">
  <div class="relative flex space-x-16 border border-red-500 py-10">
    <div class="absolute left-0 top-0 h-6 w-10 bg-blue-200">ABS</div>

    <div class="w-full bg-green-200">This should not have any space to the left</div>
  </div>
</div>
```

**Comment by wongjn** [comment]
Consider using `gap-x-*` instead:

```diff
 <div class="p-8">
-  <div class="relative flex space-x-16 border border-red-500 py-10">
+  <div class="relative flex gap-x-16 border border-red-500 py-10">
     <div class="absolute left-0 top-0 h-6 w-10 bg-blue-200">ABS</div>

     <div class="w-full bg-green-200">This should not have any space to the left</div>
   </div>
 </div>
```

**Reply by goela268** [reply]
I know about gap-x but I'm primarily using `space` in conjunction with flexbox since the support on Safari is bad for Flex + Gap.

**Comment by adamwathan** [comment]
The problem is there's just no way in CSS to even do this — you can't write a selector that says "only add margin to this element if the element before it isn't absolutely positioned".

The space utilities just use the "lobotomized owl selector" trick, and have all the same limitation — unfortunately they are not magic 😄

The best solutions are:

1. Use gap as mentioned, that's what I'd do myself.
2. Restructure your HTML to make sure only the elements you actually want to be affected by the `space-x-*` utility are inside the element that has that class — this is also a great solution.

A hacky solution that is kind of fun but I personally wouldn't recommend is to add the `hidden` attribute to the absolute positioned `div`, and then force it back to being displayed with `block`:

https://play.tailwindcss.com/hzEYnmWE6V

We ignore any elements with the `hidden` attribute when applying the `space-*` utilities but there might be important semantics it applies that mess with assistive technology when forcing it back to being displayed with `block`, not sure.
